WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadHandler.java

```diff
@@ -279,12 +266,26 @@ protected Boolean doInBackground(PostModel... posts) {
 
             RemotePostPayload payload = new RemotePostPayload(mPost, mSite);
 
-            if (sShouldRemoteAutoSavePosts.contains(mPost.getId())) {
-                mDispatcher.dispatch(PostActionBuilder.newRemoteAutoSavePostAction(payload));
-            } else {
-                mDispatcher.dispatch(PostActionBuilder.newPushPostAction(payload));
+            switch (UploadUtils.getPostUploadAction(mPost)) {
```

Comment:
This is one of the main changes in this PR. The PostUploadHandler is the place where we decide whether the post will be uploaded, uploaded as draft or remote auto saved.